### ISSUE TITLE
Base64 validation wrongly performed after decoding the string field

### ISSUE THREAD

**Issue by mtfelian** [go-swagger/go-swagger#1548]
## Problem statement

Base64 validation wrongly performed after decoding the string field. 

We found it with my colleague today after adding the `format: body` into the string field in spec.

## Swagger specification
```
swagger: '2.0'

info:
  title: My App API
  description: My App API
  version: 1.0.0
  contact:
    name: Artemy Shepelev
    email: ffelian@gmail.com

schemes:
  - http

produces:
  - application/json

consumes:
  - application/json

paths:
  /in:
    post:
      summary: My method
      operationId: MyMethod
      parameters:
      - name: data
        in: body
        schema:
          type: string
          format: byte
      responses:
        default:
          description: Ok
          schema:
            type: string
```
## Steps to reproduce

put the spec into the file `swagger.yml`
`swagger generate server -f ./swagger.yml`
`go run cmd/my-app-server/main.go --port=3000`
in separate console: 

`curl -H 'Content-Type:application/json' -X POST localhost:3000/in --data '"bWVzc2FnZWhlcmU="'`
returns `{"code":601,"message":"data in body must be of type byte: \"messagehere\""}`

but `curl -H 'Content-Type:application/json' -X POST localhost:3000/in --data '"MjIzM3QyNHQ="'`
works normally by coincidence. 

## Environment
swagger version prints "dev" (latest from the repo)
go version: 1.10.2
OS:   Ubuntu 14

**Comment by fredbi** [comment]
The problem is generic with this format.
Unlike other formats (e.g. date, uuid...), the actual validation is carried on by the unmarshaller.
The format validation in this case applies to raw strings and not already unmarshalled values.

In my opinion, we should remove the validation on this format everywhere (models, parameters).

**Comment by mtfelian** [comment]
Yes, think "to validate, try to decode". It's not good idea to validate base64 or even json by regexp.

**Comment by fredbi** [comment]
Ok. So I think that in `generator/model.go` and `generator/parameter.go`, everywhere we put the exception on format validation with IsStream (for format: binary), we should add this exception too for format bytes.
